cmd/gateway/s3/gateway-s3.go

```diff
@@ -523,11 +539,16 @@ func (l *s3Objects) ListMultipartUploads(ctx context.Context, bucket string, pre
 // NewMultipartUpload upload object in multiple parts
 func (l *s3Objects) NewMultipartUpload(ctx context.Context, bucket string, object string, o minio.ObjectOptions) (uploadID string, err error) {
 	// Create PutObject options
-	opts := miniogo.PutObjectOptions{UserMetadata: o.UserDefined, ServerSideEncryption: o.ServerSideEncryption}
+	var tags map[string]string
+	if tags, err = getTagMap(o.UserDefined["X-Amz-Tagging"]); err != nil {
+		return uploadID, minio.ErrorRespToObjectError(err, bucket, object)
```

Comment:
What sort of objecterror does this return? Have you verified that this matches with what AWS S3 returns?

Reply:
The errors are the ones that might be returned by tagging.Validate or XML decoder functions in tagging pkg. tagging.Validate() returns errors like ErrTooManyTags, ErrInvalidTag .etc. It is compatible with S3, I'd presume.